### DISCUSSION TITLE
Possible to run Rest API query from row click and have the results display in a seperate table?

### DISCUSSION THREAD

**Discussion by JClarke90** [Budibase/budibase#11758] [Help] [answered]
Hi,
Fairly new to Budibase, finding it very useful to build various tools.

I'm trying to use the results from one API query to run another API query and have thos results show in a another table, (or it can be the same table if this is makes it possible).

Here are the 2 tables, the top one has the ID that I want to use in the second API query to get the results in the table underneath it.

![image](https://github.com/Budibase/budibase/assets/12259768/6409efec-45fa-4414-956a-f422e4135135)

I have a binding that I am using in the query attached to the second table, but I don't know if its possible to have this filled in and refreshed on the click of a row in the first table

![image](https://github.com/Budibase/budibase/assets/12259768/4bf5ab91-8417-440a-8970-272da1655865)

I tried adding the following to my on row click actions but it doesn't update the second table

![image](https://github.com/Budibase/budibase/assets/12259768/41efb4ff-b4b7-44c0-b713-052435f7f859)

![image](https://github.com/Budibase/budibase/assets/12259768/e3c29772-c342-4ca8-81f2-0e7612388169)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @JClarke90,

I don't think you actually need an execute query + refresh data provider actions here. I'd recommend directly manipulating the data provider for your second table. 

1. Lets start by setting the `{{ click row.id }}` to a state value
2. Click on your initial table
3. Scroll down to `on row click` and click the actions button
4. Add the update state action, provide it with a name and give it the value of `{{ click row.id }}`
5. Save the action settings. (You can test if this is working with a paragraph component etc.)
6. Next head over to your second table
7. When you use something like a Rest query with a data provider you should see a settings cog beside it.
8. Click this cog
<img width="100%" alt="Screenshot 2023-09-14 at 08 02 56" src="https://github.com/Budibase/budibase/assets/126772285/932d9f47-dcc3-4cfb-9ea4-f7624270376d">
9. Add the newly created state variable to your id binding you created for the rest query inside the drawl thats opened.
<img width="1318" alt="Screenshot 2023-09-14 at 08 09 35" src="https://github.com/Budibase/budibase/assets/126772285/1ec3e548-a5d4-43d3-bd19-fd558bf5865f">

**Output**
![Screen Recording 2023-09-14 at 08 14 15](https://github.com/Budibase/budibase/assets/126772285/6de783b6-0ea3-4ac8-888b-5c60a0ce245c)

I hope this solves your issue!

**Reply by JClarke90** [reply]
Thank you this solved my problem, I just need to work through my last 2 issues and ill be all good.
